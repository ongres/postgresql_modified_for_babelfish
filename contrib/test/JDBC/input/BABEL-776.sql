USE master;
GO

SELECT FORMATMESSAGE('%s', 'Hi')
GO

SELECT FORMATMESSAGE('Hello %s', CAST('World' as VARCHAR(10)))
GO

SELECT FORMATMESSAGE('Testing %s, %s, %s', CAST('one' AS VARCHAR(10)), CAST('two' AS VARCHAR(10)), CAST('three' AS VARCHAR(10)))
GO

SELECT formatmessage('Testing adjacent %d%s%d%s', 1, CAST('two' as VARCHAR(10)), 3, CAST('four' AS VARCHAR(10)))
GO

SELECT formatmessage('d: %d %d %d', 1, 2, 3)
GO

SELECT formatmessage('i: %i, %i, %i', 1, 2, 3)
GO

SELECT formatmessage('Extra params with no format', CAST('asdf' AS VARCHAR(10)), CAST('asdf' AS VARCHAR(10)), CAST('asdf' AS VARCHAR(10)))
GO

select formatmessage('Not enough parameters: %s, %s, %s, %s', CAST('1' AS VARCHAR(10)), CAST('2' AS VARCHAR(10)))
GO

SELECT formatmessage('More parameters than %s', CAST('placeholders' AS VARCHAR(12)), CAST('to' AS VARCHAR(10)), CAST('handle' AS VARCHAR(10)))
GO

SELECT formatmessage('Testing no inputs with placeholder %s')
GO

SELECT FORMATMESSAGE('Unsigned hexadecimal %x, %X, %X, %X, %x', 11, 11, -11, 50, -50)
GO

SELECT FORMATMESSAGE('Unsigned int %u, %u', 50, -50)
GO

SELECT FORMATMESSAGE('Unsigned octal %o, %o', 50, -50)
GO

SELECT FORMATMESSAGE('Nonlatin: 亚马%s', CAST('逊' AS VARCHAR(10)))
GO

SELECT FORMATMESSAGE('Long string tests: \n\n Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla consequat metus sit amet lectus vulputate lacinia. Nam vel auctor enim, quis finibus lorem. Quisque maximus egestas est, et vehicula lectus. Curabitur sodales efficitur nibh, quis varius metus semper vel. Aenean quis rhoncus tortor. Nullam eleifend blandit imperdiet. Donec a gravida ex. Cras ac nulla orci. Fusce ut est ac odio consequat rutrum. Morbi diam erat, blandit ut nunc efficitur, venenatis placerat nunc. Sed laoreet aliquam leo. Quisque vestibulum hendrerit mi, at cursus eros aliquet vitae. Mauris a sem blandit, interdum ligula et, feugiat nulla. Sed quis ipsum at dolor congue vulputate at elementum ex. Integer iaculis, libero nec pulvinar molestie, neque mi vestibulum nisl, in volutpat metus arcu in purus. Praesent ullamcorper mi sit amet consectetur mattis. Sed quis ex placerat, finibus purus dignissim, porttitor nibh. Maecenas dignissim tristique tempus. Proin venenatis sem a orci sagittis, vel consequat sem vestibulum. Donec consectetur cursus tortor, eget volutpat lacus posuere a. Pellentesque eu orci hendrerit sapien dictum fringilla. Morbi rutrum mollis ipsum sit amet hendrerit. Nullam nec felis tortor. Mauris augue turpis, volutpat ac odio in, faucibus venenatis purus. Curabitur scelerisque pharetra nunc non mattis. Nunc sagittis euismod mi sit amet gravida. Integer malesuada pretium nibh. Maecenas sagittis facilisis enim, et hendrerit nulla fringilla in. Nam pulvinar, nibh non mattis hendrerit, arcu velit molestie sem, sed rutrum arcu odio a risus. Pellentesque lorem sem, dictum nec sem nec, congue ornare purus. Mauris id efficitur mi. Phasellus nec tortor lacus. Integer vitae tempus lorem. Suspendisse augue orci, volutpat vitae magna id, venenatis viverra nunc. Sed lobortis faucibus ante, ac porttitor risus. Proin viverra vestibulum enim, sed dignissim erat lobortis scelerisque. Aliquam et diam ut lacus dictum tristique. Sed a sapien quis nulla semper elementum. Ut fringilla laoreet luctus. Vivamus mi ipsum. %s', CAST('Ellipsis' AS VARCHAR(3000)))
GO

SELECT FORMATMESSAGE('Long string test parameter: %s', CAST('Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla consequat metus sit amet lectus vulputate lacinia. Nam vel auctor enim, quis finibus lorem. Quisque maximus egestas est, et vehicula lectus. Curabitur sodales efficitur nibh, quis varius metus semper vel. Aenean quis rhoncus tortor. Nullam eleifend blandit imperdiet. Donec a gravida ex. Cras ac nulla orci. Fusce ut est ac odio consequat rutrum. Morbi diam erat, blandit ut nunc efficitur, venenatis placerat nunc. Sed laoreet aliquam leo. Quisque vestibulum hendrerit mi, at cursus eros aliquet vitae. Mauris a sem blandit, interdum ligula et, feugiat nulla. Sed quis ipsum at dolor congue vulputate at elementum ex. Integer iaculis, libero nec pulvinar molestie, neque mi vestibulum nisl, in volutpat metus arcu in purus. Praesent ullamcorper mi sit amet consectetur mattis. Sed quis ex placerat, finibus purus dignissim, porttitor nibh. Maecenas dignissim tristique tempus. Proin venenatis sem a orci sagittis, vel consequat sem vestibulum. Donec consectetur cursus tortor, eget volutpat lacus posuere a. Pellentesque eu orci hendrerit sapien dictum fringilla. Morbi rutrum mollis ipsum sit amet hendrerit. Nullam nec felis tortor. Mauris augue turpis, volutpat ac odio in, faucibus venenatis purus. Curabitur scelerisque pharetra nunc non mattis. Nunc sagittis euismod mi sit amet gravida. Integer malesuada pretium nibh. Maecenas sagittis facilisis enim, et hendrerit nulla fringilla in. Nam pulvinar, nibh non mattis hendrerit, arcu velit molestie sem, sed rutrum arcu odio a risus. Pellentesque lorem sem, dictum nec sem nec, congue ornare purus. Mauris id efficitur mi. Phasellus nec tortor lacus. Integer vitae tempus lorem. Suspendisse augue orci, volutpat vitae magna id, venenatis viverra nunc. Sed lobortis faucibus ante, ac porttitor risus. Proin viverra vestibulum enim, sed dignissim erat lobortis scelerisque. Aliquam et diam ut lacus dictum tristique. Sed a sapien quis nulla semper elementum. Ut fringilla laoreet luctus. Vivamus mi ipsum.i' AS VARCHAR(3000)))
GO

SELECT FORMATMESSAGE('21 args: %s %s %s %s %s %s %s %s %s %s %s %s %s %s %s %s %s %s %s %s %s', CAST('one' AS VARCHAR(10)), CAST('two' AS VARCHAR(10)),CAST('three' AS VARCHAR(10)),CAST('four' AS VARCHAR(10)),CAST('five' AS VARCHAR(10)),CAST('six' AS VARCHAR(10)),CAST('seven' AS VARCHAR(10)),CAST('eight' AS VARCHAR(10)),CAST('nine' AS VARCHAR(10)),CAST('ten' AS VARCHAR(10)),CAST('eleven' AS VARCHAR(10)),CAST('twelve' AS VARCHAR(10)),CAST('thirteen' AS VARCHAR(10)),CAST('fourteen' AS VARCHAR(10)),CAST('fifteen' AS VARCHAR(10)),CAST('sixteen' AS VARCHAR(10)),CAST('seventeen' AS VARCHAR(10)),CAST('eighteen' AS VARCHAR(10)),CAST('nineteen' AS VARCHAR(10)),CAST('twenty' AS VARCHAR(10)),CAST('twenty-one' AS VARCHAR(10)))
GO

SELECT FORMATMESSAGE('Invalid parameter example: %s', TRUE)
GO

SELECT FORMATMESSAGE('Invalid placeholder example: %m', 0)
GO

SELECT FORMATMESSAGE(NULL, 0)
GO

SELECT FORMATMESSAGE('Placeholder null: %s', NULL)
GO

SELECT FORMATMESSAGE('Mismatch datatype: %d', 'string')
GO

SELECT FORMATMESSAGE('Mismatch datatype: %o', CAST('string' AS VARCHAR(10)))
GO

SELECT FORMATMESSAGE('Mismatch datatype: %s', 123);
GO
