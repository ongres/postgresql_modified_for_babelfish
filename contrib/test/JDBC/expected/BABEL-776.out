USE master;
GO

SELECT FORMATMESSAGE('%s', 'Hi')
GO
~~START~~
nvarchar
Hi
~~END~~


SELECT FORMATMESSAGE('Hello %s', CAST('World' as VARCHAR(10)))
GO
~~START~~
nvarchar
Hello World
~~END~~


SELECT FORMATMESSAGE('Testing %s, %s, %s', CAST('one' AS VARCHAR(10)), CAST('two' AS VARCHAR(10)), CAST('three' AS VARCHAR(10)))
GO
~~START~~
nvarchar
Testing one, two, three
~~END~~


SELECT formatmessage('Testing adjacent %d%s%d%s', 1, CAST('two' as VARCHAR(10)), 3, CAST('four' AS VARCHAR(10)))
GO
~~START~~
nvarchar
Testing adjacent 1two3four
~~END~~


SELECT formatmessage('d: %d %d %d', 1, 2, 3)
GO
~~START~~
nvarchar
d: 1 2 3
~~END~~


SELECT formatmessage('i: %i, %i, %i', 1, 2, 3)
GO
~~START~~
nvarchar
i: 1, 2, 3
~~END~~


SELECT formatmessage('Extra params with no format', CAST('asdf' AS VARCHAR(10)), CAST('asdf' AS VARCHAR(10)), CAST('asdf' AS VARCHAR(10)))
GO
~~START~~
nvarchar
Extra params with no format
~~END~~


select formatmessage('Not enough parameters: %s, %s, %s, %s', CAST('1' AS VARCHAR(10)), CAST('2' AS VARCHAR(10)))
GO
~~START~~
nvarchar
Not enough parameters: 1, 2, (null), (null)
~~END~~


SELECT formatmessage('More parameters than %s', CAST('placeholders' AS VARCHAR(12)), CAST('to' AS VARCHAR(10)), CAST('handle' AS VARCHAR(10)))
GO
~~START~~
nvarchar
More parameters than placeholders
~~END~~


SELECT formatmessage('Testing no inputs with placeholder %s')
GO
~~START~~
nvarchar
Testing no inputs with placeholder (null)
~~END~~


SELECT FORMATMESSAGE('Unsigned hexadecimal %x, %X, %X, %X, %x', 11, 11, -11, 50, -50)
GO
~~START~~
nvarchar
Unsigned hexadecimal b, B, FFFFFFF5, 32, ffffffce
~~END~~


SELECT FORMATMESSAGE('Unsigned int %u, %u', 50, -50)
GO
~~START~~
nvarchar
Unsigned int 50, 4294967246
~~END~~


SELECT FORMATMESSAGE('Unsigned octal %o, %o', 50, -50)
GO
~~START~~
nvarchar
Unsigned octal 62, 37777777716
~~END~~


SELECT FORMATMESSAGE('Nonlatin: 亚马%s', CAST('逊' AS VARCHAR(10)))
GO
~~START~~
nvarchar
Nonlatin: 亚马逊
~~END~~


SELECT FORMATMESSAGE('Long string tests: \n\n Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla consequat metus sit amet lectus vulputate lacinia. Nam vel auctor enim, quis finibus lorem. Quisque maximus egestas est, et vehicula lectus. Curabitur sodales efficitur nibh, quis varius metus semper vel. Aenean quis rhoncus tortor. Nullam eleifend blandit imperdiet. Donec a gravida ex. Cras ac nulla orci. Fusce ut est ac odio consequat rutrum. Morbi diam erat, blandit ut nunc efficitur, venenatis placerat nunc. Sed laoreet aliquam leo. Quisque vestibulum hendrerit mi, at cursus eros aliquet vitae. Mauris a sem blandit, interdum ligula et, feugiat nulla. Sed quis ipsum at dolor congue vulputate at elementum ex. Integer iaculis, libero nec pulvinar molestie, neque mi vestibulum nisl, in volutpat metus arcu in purus. Praesent ullamcorper mi sit amet consectetur mattis. Sed quis ex placerat, finibus purus dignissim, porttitor nibh. Maecenas dignissim tristique tempus. Proin venenatis sem a orci sagittis, vel consequat sem vestibulum. Donec consectetur cursus tortor, eget volutpat lacus posuere a. Pellentesque eu orci hendrerit sapien dictum fringilla. Morbi rutrum mollis ipsum sit amet hendrerit. Nullam nec felis tortor. Mauris augue turpis, volutpat ac odio in, faucibus venenatis purus. Curabitur scelerisque pharetra nunc non mattis. Nunc sagittis euismod mi sit amet gravida. Integer malesuada pretium nibh. Maecenas sagittis facilisis enim, et hendrerit nulla fringilla in. Nam pulvinar, nibh non mattis hendrerit, arcu velit molestie sem, sed rutrum arcu odio a risus. Pellentesque lorem sem, dictum nec sem nec, congue ornare purus. Mauris id efficitur mi. Phasellus nec tortor lacus. Integer vitae tempus lorem. Suspendisse augue orci, volutpat vitae magna id, venenatis viverra nunc. Sed lobortis faucibus ante, ac porttitor risus. Proin viverra vestibulum enim, sed dignissim erat lobortis scelerisque. Aliquam et diam ut lacus dictum tristique. Sed a sapien quis nulla semper elementum. Ut fringilla laoreet luctus. Vivamus mi ipsum. %s', CAST('Ellipsis' AS VARCHAR(3000)))
GO
~~START~~
nvarchar
Long string tests: \n\n Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla consequat metus sit amet lectus vulputate lacinia. Nam vel auctor enim, quis finibus lorem. Quisque maximus egestas est, et vehicula lectus. Curabitur sodales efficitur nibh, quis varius metus semper vel. Aenean quis rhoncus tortor. Nullam eleifend blandit imperdiet. Donec a gravida ex. Cras ac nulla orci. Fusce ut est ac odio consequat rutrum. Morbi diam erat, blandit ut nunc efficitur, venenatis placerat nunc. Sed laoreet aliquam leo. Quisque vestibulum hendrerit mi, at cursus eros aliquet vitae. Mauris a sem blandit, interdum ligula et, feugiat nulla. Sed quis ipsum at dolor congue vulputate at elementum ex. Integer iaculis, libero nec pulvinar molestie, neque mi vestibulum nisl, in volutpat metus arcu in purus. Praesent ullamcorper mi sit amet consectetur mattis. Sed quis ex placerat, finibus purus dignissim, porttitor nibh. Maecenas dignissim tristique tempus. Proin venenatis sem a orci sagittis, vel consequat sem vestibulum. Donec consectetur cursus tortor, eget volutpat lacus posuere a. Pellentesque eu orci hendrerit sapien dictum fringilla. Morbi rutrum mollis ipsum sit amet hendrerit. Nullam nec felis tortor. Mauris augue turpis, volutpat ac odio in, faucibus venenatis purus. Curabitur scelerisque pharetra nunc non mattis. Nunc sagittis euismod mi sit amet gravida. Integer malesuada pretium nibh. Maecenas sagittis facilisis enim, et hendrerit nulla fringilla in. Nam pulvinar, nibh non mattis hendrerit, arcu velit molestie sem, sed rutrum arcu odio a risus. Pellentesque lorem sem, dictum nec sem nec, congue ornare purus. Mauris id efficitur mi. Phasellus nec tortor lacus. Integer vitae tempus lorem. Suspendisse augue orci, volutpat vitae magna id, venenatis viverra nunc. Sed lobortis faucibus ante, ac porttitor risus. Proin viverra vestibulum enim, sed dignissim erat lobortis scelerisque. Aliquam et diam ut lacus dictum tristique. Sed a sapien quis nulla semper elementum. Ut fringilla laoreet luctus. Vivamus mi ipsu...
~~END~~


SELECT FORMATMESSAGE('Long string test parameter: %s', CAST('Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla consequat metus sit amet lectus vulputate lacinia. Nam vel auctor enim, quis finibus lorem. Quisque maximus egestas est, et vehicula lectus. Curabitur sodales efficitur nibh, quis varius metus semper vel. Aenean quis rhoncus tortor. Nullam eleifend blandit imperdiet. Donec a gravida ex. Cras ac nulla orci. Fusce ut est ac odio consequat rutrum. Morbi diam erat, blandit ut nunc efficitur, venenatis placerat nunc. Sed laoreet aliquam leo. Quisque vestibulum hendrerit mi, at cursus eros aliquet vitae. Mauris a sem blandit, interdum ligula et, feugiat nulla. Sed quis ipsum at dolor congue vulputate at elementum ex. Integer iaculis, libero nec pulvinar molestie, neque mi vestibulum nisl, in volutpat metus arcu in purus. Praesent ullamcorper mi sit amet consectetur mattis. Sed quis ex placerat, finibus purus dignissim, porttitor nibh. Maecenas dignissim tristique tempus. Proin venenatis sem a orci sagittis, vel consequat sem vestibulum. Donec consectetur cursus tortor, eget volutpat lacus posuere a. Pellentesque eu orci hendrerit sapien dictum fringilla. Morbi rutrum mollis ipsum sit amet hendrerit. Nullam nec felis tortor. Mauris augue turpis, volutpat ac odio in, faucibus venenatis purus. Curabitur scelerisque pharetra nunc non mattis. Nunc sagittis euismod mi sit amet gravida. Integer malesuada pretium nibh. Maecenas sagittis facilisis enim, et hendrerit nulla fringilla in. Nam pulvinar, nibh non mattis hendrerit, arcu velit molestie sem, sed rutrum arcu odio a risus. Pellentesque lorem sem, dictum nec sem nec, congue ornare purus. Mauris id efficitur mi. Phasellus nec tortor lacus. Integer vitae tempus lorem. Suspendisse augue orci, volutpat vitae magna id, venenatis viverra nunc. Sed lobortis faucibus ante, ac porttitor risus. Proin viverra vestibulum enim, sed dignissim erat lobortis scelerisque. Aliquam et diam ut lacus dictum tristique. Sed a sapien quis nulla semper elementum. Ut fringilla laoreet luctus. Vivamus mi ipsum.i' AS VARCHAR(3000)))
GO
~~START~~
nvarchar
Long string test parameter: Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla consequat metus sit amet lectus vulputate lacinia. Nam vel auctor enim, quis finibus lorem. Quisque maximus egestas est, et vehicula lectus. Curabitur sodales efficitur nibh, quis varius metus semper vel. Aenean quis rhoncus tortor. Nullam eleifend blandit imperdiet. Donec a gravida ex. Cras ac nulla orci. Fusce ut est ac odio consequat rutrum. Morbi diam erat, blandit ut nunc efficitur, venenatis placerat nunc. Sed laoreet aliquam leo. Quisque vestibulum hendrerit mi, at cursus eros aliquet vitae. Mauris a sem blandit, interdum ligula et, feugiat nulla. Sed quis ipsum at dolor congue vulputate at elementum ex. Integer iaculis, libero nec pulvinar molestie, neque mi vestibulum nisl, in volutpat metus arcu in purus. Praesent ullamcorper mi sit amet consectetur mattis. Sed quis ex placerat, finibus purus dignissim, porttitor nibh. Maecenas dignissim tristique tempus. Proin venenatis sem a orci sagittis, vel consequat sem vestibulum. Donec consectetur cursus tortor, eget volutpat lacus posuere a. Pellentesque eu orci hendrerit sapien dictum fringilla. Morbi rutrum mollis ipsum sit amet hendrerit. Nullam nec felis tortor. Mauris augue turpis, volutpat ac odio in, faucibus venenatis purus. Curabitur scelerisque pharetra nunc non mattis. Nunc sagittis euismod mi sit amet gravida. Integer malesuada pretium nibh. Maecenas sagittis facilisis enim, et hendrerit nulla fringilla in. Nam pulvinar, nibh non mattis hendrerit, arcu velit molestie sem, sed rutrum arcu odio a risus. Pellentesque lorem sem, dictum nec sem nec, congue ornare purus. Mauris id efficitur mi. Phasellus nec tortor lacus. Integer vitae tempus lorem. Suspendisse augue orci, volutpat vitae magna id, venenatis viverra nunc. Sed lobortis faucibus ante, ac porttitor risus. Proin viverra vestibulum enim, sed dignissim erat lobortis scelerisque. Aliquam et diam ut lacus dictum tristique. Sed a sapien quis nulla semper elementum. Ut fringilla laoreet luctus. Vivamus mi ...
~~END~~


SELECT FORMATMESSAGE('21 args: %s %s %s %s %s %s %s %s %s %s %s %s %s %s %s %s %s %s %s %s %s', CAST('one' AS VARCHAR(10)), CAST('two' AS VARCHAR(10)),CAST('three' AS VARCHAR(10)),CAST('four' AS VARCHAR(10)),CAST('five' AS VARCHAR(10)),CAST('six' AS VARCHAR(10)),CAST('seven' AS VARCHAR(10)),CAST('eight' AS VARCHAR(10)),CAST('nine' AS VARCHAR(10)),CAST('ten' AS VARCHAR(10)),CAST('eleven' AS VARCHAR(10)),CAST('twelve' AS VARCHAR(10)),CAST('thirteen' AS VARCHAR(10)),CAST('fourteen' AS VARCHAR(10)),CAST('fifteen' AS VARCHAR(10)),CAST('sixteen' AS VARCHAR(10)),CAST('seventeen' AS VARCHAR(10)),CAST('eighteen' AS VARCHAR(10)),CAST('nineteen' AS VARCHAR(10)),CAST('twenty' AS VARCHAR(10)),CAST('twenty-one' AS VARCHAR(10)))
GO
~~START~~
nvarchar
21 args: one two three four five six seven eight nine ten eleven twelve thirteen fourteen fifteen sixteen seventeen eighteen nineteen twenty (null)
~~END~~


SELECT FORMATMESSAGE('Invalid parameter example: %s', TRUE)
GO
~~ERROR (Code: 33557097)~~

~~ERROR (Message: Unsupported type with type boolean)~~


SELECT FORMATMESSAGE('Invalid placeholder example: %m', 0)
GO
~~ERROR (Code: 2787)~~

~~ERROR (Message: Invalid format specification: %m)~~


SELECT FORMATMESSAGE(NULL, 0)
GO
~~START~~
nvarchar
<NULL>
~~END~~


SELECT FORMATMESSAGE('Placeholder null: %s', NULL)
GO
~~START~~
nvarchar
Placeholder null: (null)
~~END~~


SELECT FORMATMESSAGE('Mismatch datatype: %d', 'string')
GO
~~ERROR (Code: 33557097)~~

~~ERROR (Message: Param 1 expected format type %d but received type unknown)~~


SELECT FORMATMESSAGE('Mismatch datatype: %o', CAST('string' AS VARCHAR(10)))
GO
~~ERROR (Code: 33557097)~~

~~ERROR (Message: Param 1 expected format type %o but received type "varchar")~~


SELECT FORMATMESSAGE('Mismatch datatype: %s', 123);
GO
~~ERROR (Code: 33557097)~~

~~ERROR (Message: Param 1 expected format type %s but received type integer)~~

